scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -542,11 +542,11 @@ def mark_to_warm(self, container_id, pod_namespace, pod_name):
         self.__lock.acquire()
         try:
             if not container_id in self.__active_pods:
-                self.__active_pods[container_id] = ControlledCacheWarmer.WarmingEntry(pod_namespace, pod_name,
-                                                                                      start_time)
+                warming_entry = self.__active_pods[container_id] = ControlledCacheWarmer.WarmingEntry(
+                    pod_namespace, pod_name, start_time)
             else:
                 warming_entry = self.__active_pods[container_id]
-                was_warmed = self.__update_warming_state(warming_entry, start_time)
+            was_warmed = self.__update_warming_state(warming_entry, start_time)
```

Comment:
IMO, it's a false positive to consider something as `already_warmed` if it comes through the new entry branch.  When we create that entry, we set the `is_warm` field to `False` without even checking the cache.  If we then look at updating the warming state, then if it is in the cache, we will mark it as `already_warmed`.

In my mind, `already_warm` should only count those cases where a container/pod is active and we, at some point, observe it is not in the cache.. so it is added to the list to warm... and then before this entry is retrieved via `pick_next_pod`, the pod is added to the cache by something else.

Now, I think part of this is going to come down to what we define what `already_warm` should mean.

Reply:
iiuc for new-entry brach, if the pod is concurrently warmed then it will still get scheduled for warming, but upon `__pick_next_pod`, it will then register as `already_warm` anyway right? 

So does this become an issue of when the `already_warm` recording is done (either here in `mark_to_warm` versus later)?